File: client-server/todolist/src/app/dto/TodoFilters.ts
import {Priority} from "../model/Priority";

export class TodoFilters {

  priority: Array<Priority> | null = null;
  title: string | null = null;
  description: string | null = null;
  done: boolean | null = null;
  startTime: number | null = null;
  endTime: number | null = null;
  scheduled: boolean | null = null;
  startCreationTime: number | null = null;
  endCreationTime: number | null = null;
  page: number | null = null;
  pageSize: number | null = null;
}

File: todo-server/src/main/java/com/todo/todos/dto/TodoFilters.java
package com.todo.todos.dto;

import com.todo.todos.model.Priority;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Collection;
import java.util.Date;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class TodoFilters {

    private String title;

    private String description;

    private Collection<Priority> priority;

    private Date startCreationDate;

    private Date endCreationDate;

    private Date startDate;

    private Date endDate;

    private boolean schedule;

    private Boolean done;

    private String sub;

    private Integer page;

    private Integer pageSize;
}
